homeassistant/components/eufylife_ble/sensor.py

```diff
@@ -0,0 +1,216 @@
+
+    @callback
+    def _handle_state_update(self, *args: Any) -> None:
+        """Handle state update."""
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callback."""
+        self.async_on_remove(
+            self._data.client.register_callback(self._handle_state_update)
+        )
+
+
+class EufyLifeRealTimeWeightSensorEntity(EufyLifeSensorEntity):
+    """Representation of an EufyLife real-time weight sensor."""
+
+    _attr_name = "Real-time weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the real-time weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_real_time_weight"
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        if self._data.client.state is not None:
+            return self._data.client.state.weight_kg
+        return None
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+
+class EufyLifeWeightSensorEntity(RestoreEntity, EufyLifeSensorEntity):
+    """Representation of an EufyLife weight sensor."""
+
+    _attr_name = "Weight"
+    _attr_native_unit_of_measurement = UnitOfMass.KILOGRAMS
+    _attr_device_class = SensorDeviceClass.WEIGHT
+
+    _weight_kg: float | None = None
+
+    def __init__(self, data: EufyLifeData) -> None:
+        """Initialize the weight sensor entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{data.address}_weight"
+
+    @property
+    def available(self) -> bool:
+        """Determine if the entity is available."""
+        return True
+
+    @property
+    def native_value(self) -> float | None:
+        """Return the native value."""
+        return self._weight_kg
+
+    @property
+    def suggested_unit_of_measurement(self) -> str | None:
+        """Set the suggested unit based on the unit system."""
+        if self.hass.config.units is US_CUSTOMARY_SYSTEM:
+            return UnitOfMass.POUNDS
+
+        return UnitOfMass.KILOGRAMS
+
+    @callback
+    def _handle_state_update(self, *args: Any) -> None:
+        """Handle state update."""
+        state = self._data.client.state
+        if state is not None and state.is_final:
+            self._weight_kg = state.weight_kg
+
+        super()._handle_state_update(args)
+
+    async def async_added_to_hass(self) -> None:
+        """Restore state on startup."""
+        await super().async_added_to_hass()
+
+        last_state = await self.async_get_last_state()
+        if not last_state or last_state.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
```

Comment:
```suggestion
        if not last_state or last_state.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
```

MINOR: This could be constant `set`

Reply:
Just to confirm I'm understanding correctly, are you suggesting something like this?

```python
IGNORED_STATES = set([STATE_UNAVAILABLE, STATE_UNKNOWN])
```

and then change this code to:
```python
if not last_state or last_state.state in IGNORED_STATES:
```

Reply:
Yes, you can write it as `IGNORED_STATES = {STATE_UNAVAILABLE, STATE_UNKNOWN}`

Reply:
Fixed.